tests/integration/test_broadcast.py

```diff
@@ -0,0 +1,157 @@
+            if each[0] != each[1]:
+                is_equal = False
+                err_arr = [("iters", b.index), each[0], each[1]]
+                break
+        # test reset method
+        b.reset()
+        c.reset()
+        if b.index != c.index:
+            is_equal = False
+            err_arr = [("reset", b.index), each[0], each[1]]
+        # test whether the broadcast provide views of the original array
+        for i in range(len(arr_num)):
+            arr_num[i][(0,) * arr_num[i].ndim] = 1
+            if c.iters[i][0] != arr_num[i][(0,) * arr_num[i].ndim]:
+                is_equal = False
+                err_arr = [
+                    ("view", i),
+                    c.iters[i][0],
+                    arr_num[i][(0,) * arr_num[i].ndim],
+                ]
+
+    print_msg = f"np.broadcast({sizes})"
+    assert is_equal, (
+        f"Failed, {print_msg}\n"
+        f"Attr, {err_arr[0]}\n"
+        f"numpy result: {err_arr[1]}\n"
+        f"cunumeric_result: {err_arr[2]}\n"
+        f"cunumeric and numpy shows"
+        f" different result\n"
+    )
+
+    print(f"Passed, {print_msg}")
+
+
+def _check(*args, params: list, routine: str):
+    b = getattr(np, routine)(*args)
+    c = getattr(num, routine)(*args)
+    is_equal = True
+    if isinstance(b, list):
+        for each in zip(b, c):
+            # Try to modify multiple elements in each broadcasted array
+            if not np.array_equal(each[0], each[1]):
+                is_equal = False
+                err_arr = [("iters", b.index), each[0], each[1]]
+                break
+
+    else:
+        for each in zip(b, c):
+            if isinstance(each[0], np.ndarray) and not np.array_equal(
+                each[0], each[1]
+            ):
+                is_equal = False
+            elif isinstance(each[0], tuple) and each[0] != each[1]:
+                is_equal = False
+            if not is_equal:
+                err_arr = [each[0], each[1]]
+                break
+
+    print_msg = f"np.{routine}({params})"
+    assert is_equal, (
+        f"Failed, {print_msg}\n"
+        f"numpy result: {err_arr[0]}\n"
+        f"cunumeric_result: {err_arr[1]}\n"
+        f"cunumeric and numpy shows"
+        f" different result\n"
+    )
+
+    print(f"Passed, {print_msg}")
+
+
+def gen_shapes(dim):
+    base = (dim,)
+    result = [base]
+    for i in range(1, LEGATE_MAX_DIM):
+        base = base + (1,) if i % 2 == 0 else base + (dim,)
+        result.append(base)
+    return result
+
+
+SHAPE_LISTS = {dim: gen_shapes(dim) for dim in DIM_CASES}
+
+
+# test to run broadcast  w/ different size of arryas
+@pytest.mark.parametrize("dim", DIM_CASES, ids=str)
+def test_broadcast(dim):
+    _broadcast_check(SHAPE_LISTS[dim])
+
+
+@pytest.mark.parametrize("ndim", range(2, LEGATE_MAX_DIM))
+def test_broadcast_shapes(ndim):
+    dim = DIM_CASES[0]
+    shape_list = SHAPE_LISTS[dim]
+    _check(*shape_list, params=shape_list, routine="broadcast_shapes")
```

Comment:
you are not using ndim here. did you want to do `dim=DIM_CASES[ndim]` instead?

Reply:
you are still using a single DIM from DIM_CASES for `test_broadcast_shapes`, should we test more dimensions?

Reply:
Aside from other routines, this routine doesn't do anything but just computes the outcome shape. Others create arrays and does transformation. We can run it on multiple dimensions if needed.

Reply:
46e819cdba8bddb73af8b60641117c636362fb60